src/core/reference/include/openvino/reference/reduce_prod.hpp

```diff
@@ -25,6 +25,9 @@ template <typename T>
 void reduce_prod(const T* arg, T* out, const Shape& in_shape, const AxisSet& reduction_axes) {
     const auto out_shape = util::reduce(in_shape, reduction_axes);
     std::fill(out, out + shape_size(out_shape), T(1));
+    if (shape_size(in_shape) == 0) {
```

Comment:
This check is not required all reduce algorithms will work correctly when input shape has `0` dimension. The `CoordinateTransformBasic` will do similar check and there will be no iterations in main algorithm loop

Reply:
It's not true for all reduce ops, removal of such check from `ReduceMean` leads to division by zero error. 
In my opinion it makes code more readable, because the behavior for empty tensor is clearly defined.
I understand that it can slightly affect the binary size, so I've removed this early return where not needed.
Thanks!